packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart

```diff
@@ -316,6 +317,92 @@ flutter:
     ),
   });
 
+  {
+    final List<String> inputFilePaths = <String>[];
```

Comment:
I think this should be late and initialized for every test run in a `setUp` function--or just inlined in the one test we have here. Otherwise we'll leak the mutable state of these lists between tests, right?

Reply:
These are housed within a block scope that only contains one test. Are you afraid that tests will be added to this block in the future?
If the block construct is a bit too subtle, would you prefer a `group` to be used instead (with an empty `test` description)?

Reply:
To put it another way, IMO this is mutating global state and an antipattern:

```dart
{
  final list = <Object>[];
  test("only one test...for now", () {
    list.add(0);
    expect(list.length, 1);
  });
}
```

But this is (more or less) hermetic:

```dart
group("_", () {
  late List<Object> list;
  setUp(() {
    list = <Object>[];
  });
  test("only one test...for now", () {
    list.add(0);
    expect(list.length, 1);
  });
});
```

Reply:
The issue is not the unnamed block, but the mutable state initialized once in the block.

Reply:
Since our style guide forbids `setUp` (https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#make-each-test-entirely-self-contained) you can do:

```
group("_", () {
  List<Object> init() => <Object>[];
  test("only one test...for now", () {
    final List<Object> = init();
    list.add(0);
    expect(list.length, 1);
  });
});
```

Reply:
Thanks for clarifying. I don't think your pattern can be directly applied here. The `ProcessManager` is provided as a `Context`-housed global, and the lists need to be accessed by the `FakeCommand`. Therefore, the lists would still need to be initialized outside of the test body.

However, we don't need to provide a `ProcessManager` as a global[^*] since `Environment` accepts one in its constructor, so I declared a `FakeProcessManager` within the test body and used that instead. See commit [avoid mutable objects scoped outside of test body](https://github.com/flutter/flutter/pull/151932/commits/db927b511e246d14d447df8028a68edbf3117b5d).

[^*]: We actually still do because IIRC `testUsingContext` requires providing one if a `FileSystem` is provided, but I think it's okay to provide a second one via `FakeProcessManager.empty()` here.

Reply:
LGTM